src/IOutil.f90

```diff
@@ -244,4 +251,30 @@ SUBROUTINE GET_CURRENT_DIRECTORY_internal(dir,length,status)
       IF(status == 0) dir=tmp(1:length-1) !Remove the C_NULL_CHAR
     ENDSUBROUTINE GET_CURRENT_DIRECTORY_internal
 !
+!-------------------------------------------------------------------------------
+    SUBROUTINE MAKE_DIRECTORY_string(dir,status)
+      TYPE(StringType),INTENT(IN) :: dir
+      INTEGER(SIK),INTENT(OUT),OPTIONAL :: status
+      CALL MAKE_DIRECTORY_char(CHAR(dir),status)
+    ENDSUBROUTINE MAKE_DIRECTORY_string
+!
+!-------------------------------------------------------------------------------
+    SUBROUTINE MAKE_DIRECTORY_char(dir,status)
```

Comment:
Just protect with `if(present(status))`. Bear in mind that present-ness gets forwarded to any subsequent calls that take optional arguments, so you only need to check `present` immediately before accessing an optional variable.